nixos/modules/services/networking/webhook.nix

```diff
@@ -0,0 +1,160 @@
+
+          If set, you must create this group yourself!
+        '';
+      };
+      ip = mkOption {
+        type = types.str;
+        default = "0.0.0.0";
+        description = mdDoc ''
+          The IP webhook should serve hooks on.
+
+          The default means it be reached on any interface if `openFirewall = true`.
+          Otherwise, it can only be reached from the localhost.
+        '';
+      };
+      port = mkOption {
+        type = types.port;
+        default = 9000;
+        description = mdDoc "The port webhook should be reachable from.";
+      };
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Open the configured port in the firewall for external ingress traffic.
+          Preferably the Webhook server is instead put behind a reverse proxy.
+        '';
+      };
+      enableTemplate = mkOption {
+        type = types.bool;
+        default = false;
+        description = mdDoc "Enable the generated hooks file to be parsed as a Go template.";
+      };
+      urlPrefix = mkOption {
+        type = types.str;
+        default = "hooks";
+        description = mdDoc ''
+          The URL path prefix to use for served hooks
+
+          Example: protocol://yourserver:port/PREFIX/:hook-id
+        '';
+      };
+      hooks = mkOption {
+        type = types.str;
+        default = ''
+        [{
+          "id": "echo",
+          "execute-command": "echo",
+          "response-message": "Webhook is reachable!"
+        }]'';
+        description = mdDoc ''
+          The actual configuration of which hooks will be served.
+
+          It cannot be empty otherwise your service won't start.
+          Read more on the [project homepage] and on the [hook definition] page.
+
+          [hook definition]: https://github.com/adnanh/webhook/blob/master/docs/Hook-Definition.md
+          [project homepage]: https://github.com/adnanh/webhook#configuration
+        '';
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.users = mkIf (cfg.user == defaultUser) {
+      ${defaultUser} =
+        {
+          isSystemUser = true;
+          group = cfg.group;
+          description = "Webhook daemon user";
+        };
+    };
+
+    users.groups = mkIf (cfg.user == defaultUser && cfg.group == defaultUser) {
+      ${defaultUser} = {};
+    };
+
+    networking.firewall = lib.mkIf cfg.openFirewall {
+      allowedTCPPorts = [ cfg.port ];
+    };
+
+    systemd.services = {
+      webhook = {
+        description = "Webhook service";
+        after = [ "network.target" ];
+        environment = {
+          STNORESTART = "yes";
+          STNOUPGRADE = "yes";
+          inherit (cfg);
+        } // config.networking.proxy.envVars;
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+          Restart = "on-failure";
+          SuccessExitStatus = "2 3 4";
+          RestartForceExitStatus="3 4";
```

Comment:
Is this needed or was it also cargo-culted from the Syncthing service? What do these exit statuses mean for webhook?

Reply:
Syncthing probably served as a template in the original PR.

From a cursory glance at webhook, it seems like `0` is the success exit status and `1` is the only other one that I could find: https://github.com/adnanh/webhook/search?q=exit

![image](https://user-images.githubusercontent.com/2829538/209692975-c44d2eb2-399b-4e46-8df5-a38b010037f8.png)

I'll use those

Reply:
We should just remove the three options. Those are the default exit status semantics, and there's no reason to restart on failure a priori.

Reply:
Applied in 2a9220c

Reply:
Oh, I didn't see your response. In any case, I do think that it should be restarted on failure. It's akin to a web-server and IMO those shouldn't just die forever when things go wrong.

If you like, I could introduce an option `restartOnFailure` or something similar to give users the power to decide for themselves. Or if you're fine with things as they are, you can leave this resolved.